### ISSUE TITLE
Various Enhancements/Bugfixes

### ISSUE THREAD

**Issue by matthiasblaesing** [yawkat/dbus-java#1]
Hey Jonas, I had a quick look at the bindings and liked this a lot. I have a few enhancements/bugfixes you might want to pull into your tree. If you are interested and don't want to cherry-pick/merge as a whole, please let me know. As I'm not to fond of lombok the changes might not be in your preferred style.

The changes I made:

[Decouple object location from interface](https://github.com/matthiasblaesing/dbus-java/commit/dbd9739b81387c3f89f141bcc1b9f22c09163044)

The dbus interface matches the java definition of interfaces. So what is missing is an option to specify the dbus object seperate from the interface one wants to use. I introduced an interface DBUSDestination and modified RmiFactory to allow that to be passed in with an interface definition. The interface is implemented by a basic implementation DBUSDestinationImpl.

This way a use could rewrap your bindings and introduce an invocation handler, that handles these calls dynamicly (I did not try that yet).

[Move EventloopGroup to connector level and properly close it](https://github.com/matthiasblaesing/dbus-java/commit/0fafcd7fd9a4bd68d3f653d0e40388e294fd4fcf)

The EventloopGroup, that backs the DbusConnector needs to be shutdown explicitly. In the unittests this is not visible as junit shuts execution down explizitly, but a simple java programm without an explizit System.exit() will not terminate with the current design. The change makes DbusConnector Closeable and uses the close call to shutdown the EventloopGroup there. The DbusClient is modified to chain its close() call to the backing Connector.

[Add bindings for unsigned types and ObjectPath ](https://github.com/matthiasblaesing/dbus-java/commit/4c3dc955b3badf82e39f8a7ee812bf046c133678)

While playing with DBUS introspection and ObjectManager I was faced with values, that were not decoded for the DBUS->Java direction. This commit adds the bindings.

In addition to these I cherry-picked https://github.com/ValeryDubrava/yawkat-dbus-java/commit/5c8312a064155197739bca812590512e41aee8f3 into my tree to have SetProperty support. You might in general be interested in the work of Valery: https://github.com/ValeryDubrava/yawkat-dbus-java/tree/develop

**Comment by yawkat** [comment]
Thank you, these look good on first glance, I will look at the changes in detail and test them when I have the time.

**Comment by matthiasblaesing** [comment]
I added two more fixes/enhancements to my tree:

[Listener registration methods need to return Runnable, to make disconnector accessible](https://github.com/matthiasblaesing/dbus-java/commit/8b7af01f972eb3c8c069d658ad5c11b096c767c4)

The listener support relies on returning a Runnable that decouples the listener from the bus when invoked. With the current implementation of the CallSiteBuilder the Runnable return is swallowed and can't be used.  The commit changes the return type check to allow Runnable as a return value for listener registration methods and returns the Runnable instance.

[Support Struct Binding and multiple returns](https://github.com/matthiasblaesing/dbus-java/commit/53ec951a0204613ff7e2c750feed4e6e31026da8)

Start point for this enhancement was a dbus-method, that returns multiple values. The return values look like a dbus structure (positional values, different types) and this ties multiple returns and structures together.

So there are two parts:
1. a StructBinderFactory (supported by two annotations StructMember and Struct) was added
2. the CallSiteBuilder was enhanced to detect multiple returns and handle these as if they were returned as a single structure
